go/adbc/pkg/_tmpl/driver.go.tmpl

```diff
@@ -426,6 +911,46 @@ func toCdataArray(ptr *C.struct_ArrowArray) *cdata.CArrowArray {
 	return (*cdata.CArrowArray)(unsafe.Pointer(ptr))
 }
 
+//export {{.Prefix}}ConnectionCancel
+func {{.Prefix}}ConnectionCancel(cnxn *C.struct_AdbcConnection, err *C.struct_AdbcError) (code C.AdbcStatusCode) {
+	defer func() {
+		if e := recover(); e != nil {
+			code = poison(err, "AdbcConnectionCancel", e)
+		}
+	}()
+	conn := checkConnInit(cnxn, err, "AdbcConnectionCancel")
+	if conn == nil {
+		return C.ADBC_STATUS_INVALID_STATE
+	}
+
+	conn.cancelContext()
+	return C.ADBC_STATUS_OK
+}
+
+func toStrPtr(in *C.cchar_t) *string {
+	if in == nil {
+		return nil
+	}
+
+	out := C.GoString((*C.char)(in))
+	return &out
```

Comment:
if we bump to go1.20 we could avoid the string copy potentially by doing `unsafe.String((*byte)(unsafe.Pointer(in), int(C.strlen(in)))` to get the string, if you think it matters (not sure if it's worthwhile or not)

Reply:
I would rather be safe and copy